### ISSUE TITLE
files not received completely[recvfile,sendfile]

### ISSUE THREAD

**Issue by chllcy** [Haivision/srt#1234]
I'm use the example program to send and recv file(examples/sendfile.cpp,recvfile.cpp).but the file not received completely.
the udp packet not lost.I had checked with netstat -us. no new packet receive errors.

Udp:
    774152 packets received
    4056 packets to unknown port received.
    16013 packet receive errors
    794625 packets sent
    16013 receive buffer errors
    0 send buffer errors

the log below:
[root@localhost srt]# ./sendfile
15:08:46.216922/sendfile D:SRT.c: generateSocketID: : @96419073
15:08:46.217295/sendfile D:SRT.c: CHANNEL: Bound to local address: 0.0.0.0:9000
server is ready at port: 9000
15:08:57.414025/SRT:RcvQ:w1.N:SRT.c: PASSING request from: 127.0.0.1:39399 to agent:96419073
15:08:57.414688/SRT:RcvQ:w1.N:SRT.c: Listener managed the connection request from: 127.0.0.1:39399 result:waveahand
15:08:57.414830/SRT:RcvQ:w1.N:SRT.c: PASSING request from: 127.0.0.1:39399 to agent:96419073
15:08:57.414876/SRT:RcvQ:w1 D:SRT.c: generateSocketID: : @96419072
15:08:57.415302/SRT:RcvQ:w1.N:SRT.c: HSREQ/rcv: cmd=1(HSREQ) len=12 vers=0x10401 opts=0xe4 delay=0
15:08:57.415382/SRT:RcvQ:w1.N:SRT.c: listen ret: -1 - conclusion
15:08:57.415395/SRT:RcvQ:w1.N:SRT.c: Listener managed the connection request from: 127.0.0.1:39399 result:waveahand
new connection: 127.0.0.1:39399
speed = 1417.01Mbits/sec
loss = 0pkt (0%)

-----------------------------------------------------------------------------------------

[root@localhost srt]# ./recvfile 127.0.0.1 9000 /root/go1.14.1.linux-amd64.tar.gz go.tar.gz
15:08:57.413273/recvfile D:SRT.c: generateSocketID: : @171403877
15:08:57.417616/recvfile.N:SRT.c: @171403877:Connection established to: 127.0.0.1:9000
speed = 3.56142Mbits/sec
loss = 0pkt (0%)

-----------------------------------------------------------------------------------------

[root@localhost ~]# ll go1.14.1.linux-amd64.tar.gz 
-rw-r--r--. 1 root root 123631885 4月   8 03:46 go1.14.1.linux-amd64.tar.gz
[root@localhost srt]# ll go.tar.gz 
-rw-r--r--. 1 root root 110711328 4月   9 15:08 go.tar.gz

with tar xzvf go.tar.gz error
gzip: stdin: unexpected end of file
go/src/cmd/vendor/golang.org/x/arch/arm/armasm/tables.go

I had tested for many time, for small file size,it will work fine. so what's wrong?
